src/controller/python/chip/native/__init__.py

```diff
@@ -18,6 +20,39 @@ def _AllDirsToRoot(dir):
         dir = parent
 
 
+class PyChipError(ctypes.Structure):
```

Comment:
Would it be better to raise an exception immediately on construction of this object if there is indeed an error? That way, the moment you make a call into the SDK and it returns you back a `PyChipError` object, it behaves like other Python calls would, i.e throw an exception on error.

If you want to do more stuff, you'd then catch the exception to do further clean-up.

Reply:
We cannot raise exceptions when construct this, since ChipError will be passed as arguments of callbacks.
Raising exceptions for other use cases will be implemented as a part of https://github.com/project-chip/connectedhomeip/issues/22891

Mark as resolved as discussed before.